src/coreclr/vm/appdomain.hpp

```diff
@@ -822,6 +821,7 @@ typedef FileLoadLock::Holder FileLoadLockHolder;
 #endif
 class LoadLevelLimiter
 {
+    static thread_local LoadLevelLimiter* t_currentLoadLevelLimiter;
```

Comment:
Is this only used for asserts? Should it be under `#ifdef _DEBUG`?

Reply:
Looks like it is assert-only. I'll adjust it to be debug-only

Reply:
Actually, it's used in one place, `AppDomain::LoadDomainAssembly`, so we can't make it debug-only.

Reply:
Where is it exactly? I see it only in `POSTCONDITION` that is debug-only assert.

Reply:
It's used here:

https://github.com/dotnet/runtime/blob/4d44e37d09a1cbff7c91a956d7da9ed980f9a4e5/src/coreclr/vm/appdomain.cpp#L2848-L2854